posts/julia-hackathon-v4-2023-news_1.md

```diff
@@ -26,3 +26,31 @@ During this hackathon, I managed to write the logic behind task creation and syn
 ~~~
 <img src="../../assets/images/julia_hackathon_sep_2023_iu_lr.png" alt="Profile from AMDGPU diffusion benchmark on LUMI" width="100%">
 ~~~
+
+## Combining MAGEMin and t8code through Julia - with parallelization and visualization
+*Boris Kaus, Hendrik Ranocha*
+
```

Comment:
perhaps add (I don't seem to be able to edit this):

[MAGEMin](https://github.com/ComputationalThermodynamics/MAGEMin) is the new computational thermodynamics that we released last year, ofcourse along with its Julia interface [MAGEMin_C](https://github.com/ComputationalThermodynamics/MAGEMin_C.jl). Judging from the amount if emails we receive about it, it appears to be quite popular in the community. The easiest way to use it is through its graphical user interface, which is based on MATLAB as we started developing it in our pre-Julia time. The major downside of that is that it is not freely available. As a result, using it in teaching is often tricky (with students using many different operating systems). 
We are therefore very interested to move away from MATLAB and port it fully to Julia. Over the last 6 months or so we experimented with various approaches to create GUI's in Julia (Makie, Genie, and Dash.jl) and decided that [Dash.jl](https://github.com/plotly/Dash.jl) is likely to best serve our needs. It allows creating GUIs that fully run in a webpage and can also be deployed on a central server while providing nice graphics. 

A crucial part of the whole process is to have a way to create meshes that automatically zoom in on the phase boundaries. In MATLAB we used a custom-build adaptive mesh refinement method for this. During the Hackathon, Boris and Hendrik spent time on a new prototype of [MAGEMin](https://github.com/ComputationalThermodynamics/MAGEMin), a Mineral Assemblage Gibbs Energy Minimization package written in C.
Using the existing [Julia bindings](https://github.com/ComputationalThermodynamics/MAGEMin_C.jl) and the binaries created with BinaryBuilder.jl, they combined MAGEMin with the C/C++ libraries [p4est](https://github.com/cburstedde/p4est) and [t8code](https://github.com/DLR-AMR/t8code) for adaptive mesh refinement - using the corresponding Julia bindings [P4est.jl](https://github.com/trixi-framework/P4est.jl) and [T8code.jl](https://github.com/DLR-AMR/T8code.jl) including binaries from BinaryBuilder.jl, of course. Combined with shared-memory parallelization using `Base.Threads` from Julia and visualization, the new prototype is already 1.5x faster than the mature MATLAB interface, even when both call the same C-library. 
Importantly, it is also straightforward to use triangular instead of quadrilateral elements for the mesh, or extend this to 3D (if we wish to compute pseudosections as a function of pressure, temperature and chemistry, for example).

Meanwhile, Nicolas Riel spend time as a remote hackathon participant to work on plotting the resulting meshes with Plotly. Whereas it will still take some time to build the full Dash-based GUI, many of the crucial steps have now been resolved already.
Some of these modifications (like multithreading) are scheduled to be ported in the next release of [MAGEMin_C](https://github.com/ComputationalThermodynamics/MAGEMin_C.jl).

Reply:
Great, thanks! I added it in ef65729448e57bd0ada0e8b437f6723bc5bbd1d3